### PR TITLE
File uploader

### PR DESCRIPTION
**Pull request by Gamgeee**
Add storage api
Add file uploader component
Implement logo functionality for Groups

### CODE REVIEW

**Comment by AwesomeFlax** [comment]
1. Make file uploader general component (move it to appropriate folder)
2. Extract group creating/editing view from groups list model and put in same level
3. Rename group edit/create component folder to be more descriptive, its not a details view, but rather edit

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "dependencies": {
     "@emotion/react": "^11.10.0",
     "@emotion/styled": "^11.10.0",
+    "@mui/material": "^5.10.0",
```
Comment (AwesomeFlax) [inline package.json:8]: Do you really need this one?

**Review by doskachok — DISMISSED** [review]

**Comment by AwesomeFlax** [comment]
image
1. Component name in PascalCase please
2. Since you have `groups` node, there's no need to repeat it in every sub-folder, just go with `list`, `edit`, `members`, etc.

**Comment by AwesomeFlax** [comment]
image
once again, kebab-case here

File: package.json (hunk 1 above)
Comment (apximax) [reply package.json:8]: I will replace notistack with different snackbar library to exclude mui dependency

File: src/components/FileUploader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    onFileUploaded(uploadedFileId.id);
+  }, [uploadedFileId, onFileUploaded]);
+
+  useEffect(() => {
+    onUploadingChange && onUploadingChange(isFileUploading);
+  }, [onUploadingChange, isFileUploading])
+
+  const onLogoChanged = useCallback((event: ChangeEvent<HTMLInputElement>) => {
```
Comment (apximax) [inline src/components/FileUploader/index.tsx:58]: looks like incorrect function naming. Why is it 'logo' if this is universal file uploader component?

File: src/serviceWorker.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ export function register(config) {
           );
         });
       } else {
-        // Is not localhost. Just register service worker
+        // Is not localhost. Just Register service worker
         registerValidSW(swUrl, config);
       }
     });
```
Comment (apximax) [inline src/serviceWorker.js:52]: Did you really change this? Or was it generated somehow automatically?
I mean if it was made automatically, then why? Looks like something wrong added to the repo again

File: src/components/FileUploader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  children: ReactElement<any>;
+  ref: MutableRefObject<any>;
+};
+
+export interface IFileUploaderRef {
+  uploadFile: () => void;
+}
+
+const FileUploader = forwardRef(({ onFileUploaded, onFileSelected, onUploadingChange, children }: IFileUploaderProps, ref: ForwardedRef<unknown>) => {
+  const uploaderId = useMemo(() => new Date().toString(), []);
```
Comment (apximax) [inline src/components/FileUploader/index.tsx:30]: Why do you need this?

File: src/components/FileUploader/index.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+      onFileSelected && onFileSelected(file, fileData);
+    };
+
+    reader.readAsDataURL(file);
+  }, [onFileSelected]);
+
+  return (
+    <FileUploaderWrapper>
+      <label htmlFor={uploaderId}>
```
Comment (apximax) [inline src/components/FileUploader/index.tsx:76]: What the purpose of having label here?

**Review by apximax — CHANGES_REQUESTED** [review]

**Review by apximax — APPROVED** [review]

**Pull request merged**